src/Twilio/Http/WebRequestClient.cs

```diff
@@ -44,26 +44,33 @@ public override Response MakeRequest(Request request)
             }
         }
 
-        private static Exception HandleErrorResponse(HttpWebResponse errorResponse)
+        private static TwilioException HandleErrorResponse(HttpWebResponse errorResponse, Exception innerException)
         {
             if (errorResponse.StatusCode >= HttpStatusCode.InternalServerError &&
                 errorResponse.StatusCode < HttpStatusCode.HttpVersionNotSupported)
             {
-                return new TwilioException("Internal Server error: " + errorResponse.StatusDescription);
+                return new TwilioException("Internal Server error: " + errorResponse.StatusDescription, innerException, innerException);
             }
 
-            var responseStream = errorResponse.GetResponseStream();
-            var errorReader = new StreamReader(responseStream);
-            var errorContent = errorReader.ReadToEnd();
+            try
+            {
+              var responseStream = errorResponse.GetResponseStream();
+              var errorReader = new StreamReader(responseStream);
+              var errorContent = errorReader.ReadToEnd();
+            }
+            catch (Exception e)
+            {
+              throw innerException;
```

Comment:
The `innerException` is just a `Exception`. Do we want to throw a `TwilioException` maybe? I'm not sure if it's better or not, but just a thought.